core/src/main/java/org/wildfly/plugin/core/DeploymentException.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.plugin.core;
+
+/**
+ * An exception that represents a deployment error.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class DeploymentException extends RuntimeException {
+
+    /**
+     * Creates a new deployment exception.
+     *
+     * @param message the message for the exception
+     */
+    public DeploymentException(final String message) {
+        super(message);
+    }
+
+    /**
+     * Creates a new deployment exception.
+     *
+     * @param format the {@linkplain String#format(String, Object...) format} for the message
+     * @param args   the arguments for the format
+     */
+    public DeploymentException(final String format, final Object... args) {
```

Comment:
Ah, good point. I usually just add these for convenience, but they're not really necessary.